### ISSUE TITLE
"Origin" header is disallowed

### ISSUE THREAD

**Issue by christopher-johnson** [trellis-ldp/trellis#134]
I am testing the new JDK11 httpclient. (consistent with most ajax clients), the "Origin" header is now [disallowed](http://hg.openjdk.java.net/jdk/jdk/file/26ac622a4cab/src/java.net.http/share/classes/jdk/internal/net/http/common/Utils.java#l128) in a request.  I suppose that this affects the CORSConfiguration.

**Comment by acoburn** [comment]
The `Origin` header _should_ really be disallowed in a generic HTTP client. It should only be set in a web browser context. That may have implications for the test suite under jdk11 (once gradle supports jdk11), but the test suite is using the jax-rs http client for test suite interactions, so I am not sure that this will really be an issue.

**Comment by christopher-johnson** [comment]
is it possible for a client to get CORS headers without providing an "origin" header in the request?  (I may have asked this question before, or you may already addressed it)

**Comment by acoburn** [comment]
No, CORS headers are only produced when an `Origin` header is included in the request.

**Comment by christopher-johnson** [comment]
how then could a javascript client (like for the admin interface) make a CORS request?  I tested the axios client for Vue, and it needs the `Access-Control-Allow-Origin`, but it also does not allow `Origin`.

**Comment by acoburn** [comment]
Also, for javascript interactions, you may be interested in https://github.com/rdfjs/N3.js. This would be an alternate to a pure JSON-LD approach.

**Comment by acoburn** [comment]
FYI, I have a very rudimentary react-based front end application for Trellis. It also makes use of N3 and redux. There is a lot more to do before I’m ready to make the code public, but it definitely requires the presence of these CORS headers. And so far the headers appear to be working properly.

**Comment by christopher-johnson** [comment]
Great.  It works, I have also tested this with a very basic JS front end.  I guess that this issue can be closed, unless you prefer to keep it open to track any future jax.ws.rs changes.
